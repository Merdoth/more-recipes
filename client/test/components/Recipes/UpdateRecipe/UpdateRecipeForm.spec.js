import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { UpdateRecipeForm } from
  '../../../../components/Recipes/UpdateRecipe/UpdateRecipeForm.jsx';

/* global jest */

const setup = () => {
  const props = {
    match: {
      params: () => { }
    },
    user: {
      id: 1
    },
    recipe: () => { },
    message: () => { },
    goToRecipes: () => jest.fn(),
    getOneRecipe: () => { },
    updateRecipe: () => Promise.resolve()
  };
  return shallow(<UpdateRecipeForm {...props} />);
};

describe('UpdateRecipePage Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });
});
describe('componentWillReceiveProps()', () => {
  xit('should call componentWillReceiveProps()', () => {
    const wrapper = setup();
    wrapper.setProps({
      recipeName: 'fish pepper soup',
      description: 'cool fool for cool peeps',
      ingredients: 'fish and meat',
      preparation: 'cook for five mins',
      image: 'fish/jpg',
      error: ''
    });
    expect(wrapper.state('recipeName')).toEqual('fish pepper soup');
  });
});
describe('onSubmit()', () => {
  it('should update recipe to the state', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = setup();
    wrapper.setState({
      image: {}
    });
    const button = wrapper.find('#update');
    button.simulate('click', event);
  });
});
describe('onChange()', () => {
  it('should set recipeName to state when input values changes', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'recipeName',
        value: 'cool meal'
      }
    };
    const wrapper = setup();
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state.recipeName).toBe('cool meal');
  });
  it('should set image to state when input values changes', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'image',
        files: [{}]
      }
    };
    const wrapper = setup();
    wrapper.instance().onImageChange(event);
  });
});
