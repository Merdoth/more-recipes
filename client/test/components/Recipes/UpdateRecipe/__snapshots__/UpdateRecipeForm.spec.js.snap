// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateRecipePage Component snapshot it should render the right amount of elements 1`] = `
<form
  className="form-signin"
  onSubmit={[Function]}
>
  <h2
    className="form-signin-heading"
  >
    Edit Recipe
  </h2>
  <hr />
  <InputField
    label="Name"
    name="recipeName"
    onChange={[Function]}
    placeholder="Name"
    required={true}
    type="text"
    value=""
  />
  <TextArea
    label="Description"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <TextArea
    label="Ingredients"
    name="ingredients"
    onChange={[Function]}
    placeholder="Ingredients"
    type="text"
    value=""
  />
  <TextArea
    label="Preparation"
    name="preparation"
    onChange={[Function]}
    placeholder="Preparation"
    type="text"
    value=""
  />
  <div
    className="form-group"
  >
    <label
      htmlFor="foodImage"
    >
      Select Image
    </label>
    <input
      className="form-control-file"
      id="foodImage"
      name="image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <Button
    className="btn btn-lg btn-primary btn-block"
    iconClass="fa-cutlery"
    id="update"
    name="Update Recipe"
    onClick={[Function]}
    type="submit"
  />
</form>
`;
